src/Bicep.Cli/Commands/BuildCommand.cs

```diff
@@ -18,33 +20,38 @@ public class BuildCommand : ICommand
         private readonly CompilationService compilationService;
         private readonly CompilationWriter writer;
         private readonly ParametersWriter paramsWriter;
+        private readonly IFeatureProviderFactory featureProviderFactory;
 
         public BuildCommand(
             ILogger logger,
             IDiagnosticLogger diagnosticLogger,
             InvocationContext invocationContext,
             CompilationService compilationService,
             CompilationWriter writer,
-            ParametersWriter paramsWriter)
+            ParametersWriter paramsWriter,
+            IFeatureProviderFactory featureProviderFactory)
         {
             this.logger = logger;
             this.diagnosticLogger = diagnosticLogger;
             this.invocationContext = invocationContext;
             this.compilationService = compilationService;
             this.writer = writer;
             this.paramsWriter = paramsWriter;
+            this.featureProviderFactory = featureProviderFactory;
         }
 
         public async Task<int> RunAsync(BuildArguments args)
         {
             var inputPath = PathHelper.ResolvePath(args.InputFile);
+            var features = featureProviderFactory.GetFeatureProvider(PathHelper.FilePathToFileUrl(inputPath));
+            var emitterSettings = new EmitterSettings(features);
 
-            if (invocationContext.EmitterSettings.EnableSymbolicNames)
+            if (emitterSettings.EnableSymbolicNames)
```

Comment:
As a follow-up item, I think we should converge on a single message that covers enablement of all experimental features.